apis/schema/v1alpha1/mongodbdatabase_webhook.go

```diff
@@ -68,40 +68,50 @@ func (in *MongoDBDatabase) ValidateUpdate(old runtime.Object) error {
 	path := field.NewPath("spec")
 	oldDb := old.(*MongoDBDatabase)
 
-	// if phase is 'Successful', do not give permission to change the DatabaseConfig.Name
-	if oldDb.Status.Phase == DatabaseSchemaPhaseSuccessful && oldDb.Spec.Database.Config.Name != in.Spec.Database.Config.Name {
-		allErrs = append(allErrs, field.Invalid(path.Child("database").Child("config"), in.Name, `you can't change the Database Config name now`))
+	// if phase is 'Current', do not give permission to change the DatabaseConfig.Name
+	if oldDb.Status.Phase == DatabaseSchemaPhaseCurrent && oldDb.Spec.Database.Config.Name != in.Spec.Database.Config.Name {
+		allErrs = append(allErrs, field.Invalid(path.Child("database").Child("config"), in.Name, MongoDBValidateDatabaseNameChangeError))
 		return apierrors.NewInvalid(in.GroupVersionKind().GroupKind(), in.Name, allErrs)
 	}
 
-	// If Initialized==false, Do not give permission to set it to true directly
-	if oldDb.Spec.Init != nil && !oldDb.Spec.Init.Initialized && in.Spec.Init != nil && in.Spec.Init.Initialized {
-		allErrs = append(allErrs, field.Invalid(path.Child("init").Child("initialized"), in.Name, `cannot set the initialized field to true directly`))
+	// If Initialized==true, Do not give permission to set it to false directly
+	if oldDb.Spec.Init != nil && oldDb.Spec.Init.Initialized && in.Spec.Init != nil && !in.Spec.Init.Initialized {
+		allErrs = append(allErrs, field.Invalid(path.Child("init").Child("initialized"), in.Name, MongoDBValidateInitializedUnsetError))
 		return apierrors.NewInvalid(in.GroupVersionKind().GroupKind(), in.Name, allErrs)
 	}
```

Comment:
What if user applies without init?

Reply:
User applied without init, means Initialized is not true. So nothing to check here actually.  Because, I have checked this, so that user cant set intialized from true to false.

Reply:
I mean init is already true, but user applied without init again. So, `in.spec.init` will be nil

Reply:
Resolved using the following way : if oldDb.Spec.Init != nil && oldDb.Spec.Init.Initialized { 
		if in.Spec.Init == nil || (in.Spec.Init != nil && !in.Spec.Init.Initialized) {
                           // 
		}
	}

Reply:
Resolved using the following way : if oldDb.Spec.Init != nil && oldDb.Spec.Init.Initialized { 
		if in.Spec.Init == nil || (in.Spec.Init != nil && !in.Spec.Init.Initialized) {
                           // 
		}
	}